File: cdmsbats/GPSRecord.py
# -*- coding: utf-8 -*-
"""
Created on Fri Apr 11 12:14:28 2014

@author: tdoughty1
"""

from DataRecord import DataRecord


class GPSRecord(DataRecord):

    def _PrintLine(self):
        print "Found GPS Record"

    def _InitValues(self):
        pass

    def StoreValues(self, Record):

        print "Error in GPSRecord:"
        print "Should not reach this point"
        pass

    def PrintValues(self):
        pass

File: cdmsbats/gpsrecord.cpp
#include "gpsrecord.h"

GPSRecord::GPSRecord()
{
    _Date = NULL;
    _Time = NULL;
    _SubTime = NULL;
}

void GPSRecord::StoreValues(int32_t* Record)
{
    _Date = Record[0];
    _Time = Record[1];
    _SubTime = Record[2];
}

void GPSRecord::PrintValues()
{
    cout << "GPS Date = " << _Date << endl;
    cout << "GPS Time = " << _Time << endl;
    cout << "GPS SubTime = " << _SubTime << endl;
}
